### ISSUE TITLE
No local or global views in the Minimal version

### ISSUE THREAD

**Issue by lukas-kuhn** [rbalestr-lab/lejepa#11]
I am a bit confused as to why there are no global or local views in the minimal version (maybe for simplicity?) and what the sentence `For
non-ViT architectures (e.g., ResNet), set global_views = all_views.` means in the paper, since ResNets are able to handle different sizes, so why skip the global/local view step there?

Is the actual implementation closer to something like this?

```python
emb_global = model(all_views[:2])   # (2, 3, 224, 224) → (2, embed_dim)
emb_local = model(all_views[2:])    # (6, 3, 96, 96) → (6, embed_dim)
a_emb = torch.cat([emb_global, emb_local])  # (8, embed_dim)

centers = emb_global.mean(0)

sim = (centers - a_emb).square().mean()
sigreg = mean(SIGReg(emb) for emb in a_emb)
return (1-lambd)*sim + lambd*sigreg
```

**Comment by RandallBalestriero** [comment]
Yes, in the minimal code I did not include that for simplicity (so that you can easily swap any backbone). To implement it, you only need to create two RRC augmentations, one with 224 resolution, the other with say 96 resolution. Then you can forward separately the global and local views as mentioned in the paper and compute the losses accordingly. The main benefit is a reduction in GPU memory and faster training (as 6 out of the 8 views are now much lower resolution).

Also great point on resnets! In theory they are able to handle different sizes, it would be interesting to see how performances vary when using 96 and 224 for resnets instead of all 224. My guess is that this may still create some odd convolutions in the later blocks when the resolution is too small to provide meaningful gradients, but if you run those ablations and report some numbers, I would be happy to mention them in the repo!
